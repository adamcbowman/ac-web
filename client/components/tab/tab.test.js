import React from 'react'
import Renderer from 'react-test-renderer'
import { TabSet, Tab } from './'

// properties needed by the tabset component
function createNodeMock(element) {
    if (element.type === 'ul') {
        return {
            offsetWidth: 150,
            childNodes: [],
        }
    }

    return null
}

test('tab component', () => {
    const tabs = Renderer.create(
        <TabSet>
            <Tab title="Title #1">Content #1</Tab>
            <Tab title="Title #2">Content #2</Tab>
            <Tab title="Title #3">Content #3</Tab>
        </TabSet>,
        { createNodeMock }
    )

    expect(tabs).toMatchSnapshot()
})
