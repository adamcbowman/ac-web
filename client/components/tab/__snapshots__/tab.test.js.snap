// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tab component 1`] = `
<div>
  <ul
    className="List List--Compact"
    role="tablist"
  >
    <li
      className="ListItem--Active"
      onClick={[Function]}
      role="tab"
      style={null}
    >
      Title #1
      <button
        className="ExpandButton Incognito IconOnly"
        onClick={[Function]}
        type="button"
      >
        <div
          className="IconContainer"
        >
          <svg
            className={undefined}
            fill="none"
            height={24}
            viewBox="0 0 24 24"
            width={24}
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              fill="white"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </div>
      </button>
    </li>
    <li
      className="ListItem"
      onClick={[Function]}
      role="tab"
      style={null}
    >
      Title #2
      <button
        className="ExpandButton Incognito IconOnly"
        onClick={[Function]}
        type="button"
      >
        <div
          className="IconContainer"
        >
          <svg
            className={undefined}
            fill="none"
            height={24}
            viewBox="0 0 24 24"
            width={24}
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              fill="white"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </div>
      </button>
    </li>
    <li
      className="ListItem"
      onClick={[Function]}
      role="tab"
      style={null}
    >
      Title #3
      <button
        className="ExpandButton Incognito IconOnly"
        onClick={[Function]}
        type="button"
      >
        <div
          className="IconContainer"
        >
          <svg
            className={undefined}
            fill="none"
            height={24}
            viewBox="0 0 24 24"
            width={24}
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              fill="white"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </div>
      </button>
    </li>
  </ul>
  <div
    className="Panel--Active"
    role="tabpanel"
  >
    Content #1
  </div>
</div>
`;
